src/GraphQL/Types/ResolveFieldContext.cs

```diff
@@ -106,6 +108,27 @@ public bool HasArgument(string argumentName)
         {
             return Arguments?.ContainsKey(argumentName) ?? false;
         }
+
+        public async Task<object> TryAsyncResolve(Func<ResolveFieldContext<TSource>, Task<object>> resolve, Func<ExecutionErrors, Task<object>> error = null)
+        {
+            try
+            {
+                var result = await resolve(this);
+                return Task.FromResult(result);
```

Comment:
I think can just return `result` here.

Reply:
Thanks use to not wrap it but it was not working right. I wlll test again.